pkg/tcl/checktcl/subscription.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2024 Testkube.
+//
+// Licensed as a Testkube Pro file under the Testkube Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     https://github.com/kubeshop/testkube/blob/main/licenses/TCL.txt
+
+package checktcl
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+
+	"github.com/kubeshop/testkube/internal/config"
+	"github.com/kubeshop/testkube/pkg/cloud"
+	cloudconfig "github.com/kubeshop/testkube/pkg/cloud/data/config"
+	"github.com/kubeshop/testkube/pkg/cloud/data/executor"
+)
+
+type SubscriptionChecker struct {
+	proContext config.ProContext
+	orgPlan    *OrganizationPlan
+}
+
+// NewCLISubscriptionChecker creates a new subscription checker using a user token instead of the agent token
```

Comment:
is this for running from testkube cli, because it might not be connected via http?

Reply:
is it requesting cloud/enterprise api or agent api?

Reply:
Is it actually needed on the CLI level? Shouldn't it be rather indicated by API/Agent response, ending up with `402 Payment Required` or `403 Forbidden`?

Reply:
> yes, if we need it, then we should have agent api method, because it can be used also in Dashboard

In the Dashboard we already have an organization details, incl. plan (and features based on that), so the only problem is direct API access

Reply:
Should I remove it for now, and we can add it if/when we need it? For now, the only use case of the CLI connecting directly to the cloud API server is on agent setup.

Reply:
I removed this function. Let's keep Pro API communication in the agent unless there is a very good reason not to.